nucypher/network/server.py

```diff
@@ -213,20 +221,31 @@ def set_policy(self, id_as_hex, request: Request):
 
     def reencrypt_via_rest(self, id_as_hex, request: Request):
         from nucypher.policy.models import WorkOrder  # Avoid circular import
-        id = binascii.unhexlify(id_as_hex)
-        work_order = WorkOrder.from_rest_payload(id, request.body)
+        arrangement_id = binascii.unhexlify(id_as_hex)
+        work_order = WorkOrder.from_rest_payload(arrangement_id, request.body)
         self.log.info("Work Order from {}, signed {}".format(work_order.bob, work_order.receipt_signature))
         with ThreadedSession(self.db_engine) as session:
-            kfrag_bytes = self.datastore.get_policy_arrangement(id.hex().encode(),
-                                                                session=session).k_frag  # Careful!  :-)
+            policy_arrangement = self.datastore.get_policy_arrangement(arrangement_id=id_as_hex.encode(),
+                                                                       session=session)
+
+        kfrag_bytes = policy_arrangement.kfrag  # Careful!  :-)
+        verifying_key_bytes = policy_arrangement.alice_pubkey_sig.key_data
+
         # TODO: Push this to a lower level.
         kfrag = KFrag.from_bytes(kfrag_bytes)
+        alices_verifying_key = UmbralPublicKey.from_bytes(verifying_key_bytes)
         cfrag_byte_stream = b""
 
-        for capsule in work_order.capsules:
+        for capsule, capsule_signature in zip(work_order.capsules, work_order.capsule_signatures):
+            # This is the capsule signed by Bob
+            capsule_signature = bytes(capsule_signature)
+            # Ursula signs on top of it. Now both are committed to the same capsule.
+            capsule_signed_by_both = bytes(self._stamp(capsule_signature))
+
+            capsule.set_correctness_keys(verifying=alices_verifying_key)
             # TODO: Sign the result of this.  See #141.
-            cfrag = pre.reencrypt(kfrag, capsule)
-            self.log.info("Re-encrypting for Capsule {}, made CFrag {}.".format(capsule, cfrag))
+            cfrag = pre.reencrypt(kfrag, capsule, metadata=capsule_signed_by_both)
+            self.log.info("Re-encrypting for {}, made {}.".format(capsule, cfrag))
```

Comment:
```suggestion
            self.log.debug("Re-encrypting for {}, made {}.".format(capsule, cfrag))
```

Reply:
I think we should defer these changes to a later PR that addresses #485